stdlib/json.gr

```diff
@@ -2069,3 +2083,242 @@ provide let parse: (str: String) => Result<Json, JsonParseError> = (str: String)
     }
   }
 }
+
+/**
+ * Extract the value of a JSON boolean value, or `Err` if the JSON value is not
+ * a boolean.
+ * 
+ * @param json: The JSON value
+ * @returns `Ok(value)` containing the JSON boolean value or `Err(err)` if the JSON value is not a boolean
+ * 
+ * @example assert Json.boolean(Json.JsonBoolean(true)) == Ok(true)
+ * @example assert Json.boolean(Json.JsonNumber(123)) == Err(Json.IncorrectType)
+ * 
+ * @since v0.6.7
+ */
+provide let boolean = json => {
```

Comment:
I feel like we should put these in a submodule named search like you have done with the tests.